src/lib.rs

```diff
@@ -312,15 +322,15 @@ impl Trajectory for XTCTrajectory {
                 unsafe {
                     let path = path_to_cstring(&self.handle.path)?;
                     let path_p = path.into_raw();
-                    let code =
-                        xdrfile_xtc::read_xtc_natoms(path_p, &mut num_atoms as *const i32) as u32;
+                    let code = xdrfile_xtc::read_xtc_natoms(path_p, &mut num_atoms as *const i32);
                     // Reconstitute the CString so it is deallocated correctly
                     let _ = CString::from_raw(path_p);
 
                     if let Some(err) = check_code(code, ErrorTask::ReadNumAtoms) {
                         Err(err)
                     } else {
-                        Ok(num_atoms as usize)
+                        Ok(usize::try_from(num_atoms)
+                            .expect("Number of atoms in file does not fit in usize"))
```

Comment:
This casts i32 -> usize. The only reason where this can ever file is when the returned number of atoms is negative, which is a value that does not make sense in this context. Maybe we can reflect this with a more explanatory error message. Something along the lines of 
```rust
expect("Encountered illegal value for number of atoms in trajectory file: Failed to cast {} to usize.", num_atoms) 
```

Also, shouldn't this be mapped to an `Error::CouldNotCheckNAtoms`?

Reply:
The failure mode I was thinking of was on 16 bit systems, and I didn't think it was worth supporting them with a whole error variant, hence the expect. I hadn't considered that the value could be negative; I have now mapped it to a new out of range error (like `StepSizeOutOfRange`)

I don't think this is appropriate as `Error::CouldNotCheckNAtoms`, which is a variant intended specifically for `Trajectory::read()`